### ISSUE TITLE
Incompatible with units

### ISSUE THREAD

**Issue by call-a3** [terkel/mathsass#4]
Either I'm misunderstanding the usage of these functions, or they are unusable with units such as em or px. I tried the following:

```
$side: 2em;
width: sqrt(pow($side, 2)/2);
```

which fails with the following error: _node_modules/mathsass/dist/functions/pow:13: error: cannot add or subtract numbers with incompatible units_

**Comment by xi** [comment]
I don't think this is possible. Sass can only express units with whole exponents, e.g. `1em * 1em / 5px`. There is no way to express `sqrt(2em)`, `log(2em)` or `pow(2em, 0.123)`. (I think this is part of the reason why these functions are not in Sass itself)

I think it is possible in many cases to restructure the formular so this is actually not required. For example, the original example can be converted to this:

```scss
$side: 2em;
width: (abs($side) / sqrt(2));
```

Also note that the current implementations are focused on floating point numbers. This means that they are good estimations in general, but there is no guarantee that `pow(pow(2em, 8), 0.5) == pow(2em, 4)`. It might be `pow(2em, 4.00001)`.

Finally, I was interested about the current behavior:

- `pow(2em, $x)`: `pow(2, $x) * pow(1em, floor($x))`
- `sqrt(2em)`: crash
- `cos(2em)`: crash
- `log(2em)`: `log(2)`
- `acos(0.5em)`: crash

**Comment by airqb** [comment]
btw wrong Result of cos(180)*100 must return **-100** but it is returns **"74.69988"**  ??!!!

**Comment by xi** [comment]
> btw wrong Result of cos(180)*100 must return -100 but it is returns "74.69988" ??!!!

If no unit is provided, `rad` is assumed. `cos(180deg)*100` should do the trick.

**Comment by airqb** [comment]
no unit is provided.
